### PR TITLE
Update equilibria in `desc.examples`

### LINKED ISSUE #1185: Some examples do not run as intended in `desc/examples`
- Should we add the examples to the CI, to run them and ensure the results are nested and match some reference result? 
  - Or add a flag for `test_examples` we can add to a PR like `test_jax`, required to merge BUT does not run automatically so must be added each time.

### LINKED ISSUE #1186: Check + Update old notebooks on a recurring basis
- [ ] make a recurring issue to check notebooks for syntax/up-to-dateness with master and current API
- [ ] add assert statements to notebooks for correctness so nbtests also check result correctness

### PR DESCRIPTION
**Pull request by f0uriest**
Resolves #1185 
Resolves #1186 

Re-ran them all and manually verified they look good by eye and the usual metrics (low force error etc).

### CODE REVIEW

**Comment by dpanici** [comment]
This might change/need checking again once #1073 is in (or maybe base this off of #1073?)

**Comment by dpanici** [comment]
I will approve this once merge conflict is resolved, but I also want to note that I'd like each notebook to be re-ran as well. The 3D plot-containing ones may have to be each separate PRs to avoid the issues we have with the git diffs being too large and failing our tests. I will make an issue for this

**Comment by dpanici** [comment]
Could try to reduce test time as well #914 by reducing res or not doing continuation for some of the examples

**Review by unalmis — APPROVED** [review]

File: tests/baseline/test_section_F.png
Comment (YigitElma) [inline tests/baseline/test_section_F.png:1]: Is this normal? Looks weirdly uniform to me.

File: desc/examples/ATF
Comment (YigitElma) [inline desc/examples/ATF:1]: I think we can also change the radial resolution of ATF. Also, I am pretty sure this input file has a problem like #1346, because the first step will use L=6 and M=12 which will automatically set L=18, but the second step will solve it for L=M=12. I will check it tho

Comment (f0uriest) [reply desc/examples/ATF:1]: what do you mean it will set L=18? Pretty sure for the manual continuation method it should use exactly what you tell it (6 in this case)

File: tests/baseline/test_section_F.png
Comment (f0uriest) [reply tests/baseline/test_section_F.png:1]: yeah it just means theres more variation at the boundary so the middle looks relatively flat

File: desc/examples/ATF
Comment (YigitElma) [reply desc/examples/ATF:1]: This maybe a different issue than the one I mentioned but if I run this,
```python
eqfam = desc.io.load("../desc/examples/ATF_output.h5")
for eq in eq_fam:
    print(eq)
    print(np.max(eq.R_basis.modes[:,0]))
    print(np.max(eq.R_basis.modes[:,1]))
```
I get
```
Equilibrium at 0x707624133770 (L=6, M=12, N=0, NFP=12, sym=True, spectral_indexing=fringe)
15
12
Equilibrium at 0x707624826150 (L=12, M=12, N=0, NFP=12, sym=True, spectral_indexing=fringe)
18
12
Equilibrium at 0x707625c04350 (L=12, M=12, N=0, NFP=12, sym=True, spectral_indexing=fringe)
18
12
Equilibrium at 0x7076cc258350 (L=12, M=12, N=0, NFP=12, sym=True, spectral_indexing=fringe)
18
12
Equilibrium at 0x70761be6bf80 (L=12, M=12, N=2, NFP=12, sym=True, spectral_indexing=fringe)
18
12
```

Comment (dpanici) [reply desc/examples/ATF:1]: This is due to how the fringe indexing works and the fact that `L` is the max difference btwn `m` and `l` for a given mode, and also is the maximum radial-only basis degree. It is weird though yes and is a bit unintuitive esp. when L < M

Comment (YigitElma) [reply desc/examples/ATF:1]: > what do you mean it will set L=18? Pretty sure for the manual continuation method it should use exactly what you tell it (6 in this case)

By L, I was actually meaning max l mode in the basis. And if the spectral indexing is ansi, then I think you get max l 18 for L=6 and M=12.

Comment (f0uriest) [reply desc/examples/ATF:1]: yeah, `L` for a basis/equilibrium is not actually the maximum value of the radial mode number. As dario mentioned its a bit trickier with the zernike indexing. 

In any event, it has nothing to do with the continuation method, you get the same result if you just create a basis with a given L,M

```python
basis = desc.basis.ZernikePolynomial(L=6, M=12, spectral_indexing="ansi")
desc.plotting.plot_basis(basis);
```
image

Comment (YigitElma) [reply desc/examples/ATF:1]: Yeah, I don't know why but I thought there was a conditional inside modes creation like L<M and L>M. that's why I made that comment. Yes, this version works. Sorry

File: desc/equilibrium/utils.py
Comment (dpanici) [inline desc/equilibrium/utils.py:144]: what is this doing? seems to be passing in the same modes for Rn as Zn?

Comment (dpanici) [reply desc/equilibrium/utils.py:144]: That seems off? or do we ensure before this that we have the full mode spectrum in `n` and just set the non-symmetric modes to zero if it is stell symmetric?

**Review by dpanici — DISMISSED** [review]

**Review by YigitElma — DISMISSED** [review]
I think we can also reduce the radial resolution of ATF and HELITRON (also change some solver tolerances) to get the same equilibrium, as you did here for W7-X. Those are the simplest examples but have the highest radial resolution. It would help reducing the CI testing time too.

**Comment by YigitElma** [comment]
I found these to be also working. For `ATF`,
```
# spectral resolution
L_rad  =  8, 16
M_pol  =  12
N_tor  =  0x4, 4
M_grid =  12, 16
N_grid =  0x4, 6
```
For `HELIOTRON`,
```
# spectral resolution
L_rad  =  8, 18
M_pol  =  6, 12
N_tor  =  0,  0,  3
M_grid =  12, 18
N_grid =  0,  0,  6
```

**Comment by f0uriest** [comment]
> I found these to be also working. For `ATF`,
> 
> ```
> # spectral resolution
> L_rad  =  8, 16
> M_pol  =  12
> N_tor  =  0x4, 4
> M_grid =  12, 16
> N_grid =  0x4, 6
> ```
> 
> For `HELIOTRON`,
> 
> ```
> # spectral resolution
> L_rad  =  8, 18
> M_pol  =  6, 12
> N_tor  =  0,  0,  3
> M_grid =  12, 18
> N_grid =  0,  0,  6
> ```

Reducing the resolution causes some tests to fail so I'm keeping the old one for now. You can make a new PR for those if you want.

**Review by dpanici — APPROVED** [review]

**Review by YigitElma — APPROVED** [review]

**Comment by YigitElma** [comment]
> > I found these to be also working. For `ATF`,
> > 
> > ```
> > # spectral resolution
> > L_rad  =  8, 16
> > M_pol  =  12
> > N_tor  =  0x4, 4
> > M_grid =  12, 16
> > N_grid =  0x4, 6
> > ```
> > 
> > For `HELIOTRON`,
> > 
> > ```
> > # spectral resolution
> > L_rad  =  8, 18
> > M_pol  =  6, 12
> > N_tor  =  0,  0,  3
> > M_grid =  12, 18
> > N_grid =  0,  0,  6
> > ```
> 
> Reducing the resolution causes some tests to fail so I'm keeping the old one for now. You can make a new PR for those if you want.

Thanks for trying🙏

**Pull request merged**
